File: [23.12]Design-Pattern/Java/src/structural/flyweight/Main.java
package structural.flyweight;

public class Main {
    public static void main(String[] args) {
        Tree.Factory.getInstance("소나무");
        Tree.Factory.getInstance("소나무");
        Tree.Factory.getInstance("소나무");

        System.out.println();

        Tree.Factory.getInstance("대나무");
        Tree.Factory.getInstance("대나무");
        Tree.Factory.getInstance("대나무");

        System.out.println();

        Tree.Factory.getInstance("벚꽃 나무");
        Tree.Factory.getInstance("벚꽃 나무");
        Tree.Factory.getInstance("벚꽃 나무");

    }
}

File: [23.12]Design-Pattern/Kotlin/src/main/kotlin/structural/flyweight/Main.kt
package structural.flyweight

fun main() {
    Tree.Factory.getInstance("대나무")
    Tree.Factory.getInstance("대나무")
    Tree.Factory.getInstance("대나무")

    println()
    Tree.Factory.getInstance("소나무")
    Tree.Factory.getInstance("소나무")
    Tree.Factory.getInstance("소나무")

    println()
    Tree.Factory.getInstance("감나무")
    Tree.Factory.getInstance("감나무")
    Tree.Factory.getInstance("감나무")
}